packages/zoe/src/contractSupport/ratio.js

```diff
@@ -40,20 +40,17 @@ const PERCENT = 100n;
 const ratioPropertyNames = ['numerator', 'denominator'];
 
 export const assertIsRatio = ratio => {
-  assert.equal(passStyleOf(ratio), 'copyRecord');
-  const propertyNames = Object.getOwnPropertyNames(ratio);
-  assert(
-    propertyNames.length === 2,
-    X`Ratio ${ratio} must be a record with 2 fields.`,
-  );
-  for (const name of propertyNames) {
+  assertRecord(ratio, 'ratio');
+  const keys = Object.keys(ratio);
+  assert(keys.length === 2, X`Ratio ${ratio} must be a record with 2 fields.`);
+  for (const name of keys) {
     assert(
       ratioPropertyNames.includes(name),
       X`Parameter must be a Ratio record, but ${ratio} has ${q(name)}`,
     );
   }
-  Nat(ratio.numerator.value);
-  Nat(ratio.denominator.value);
+  natMathHelpers.doCoerce(ratio.numerator.value);
+  natMathHelpers.doCoerce(ratio.denominator.value);
```

Comment:
Although it works in this case, in general `doCoerce` is a coercer, not a validator. In general, only the result is guaranteed to satisfy the constraints, not the input. But we happen to know it is ok for nat. If you keep this pattern, it is at least worth an explanatory comment.

Reply:
I changed this to use `isNat` directly.